waku/v2/protocol/waku_noise/noise.nim

```diff
@@ -15,30 +14,65 @@ import bearssl
 import nimcrypto/[utils, sha2, hmac]
 
 import libp2p/stream/[connection]
+import libp2p/peerid
+import libp2p/peerinfo
 import libp2p/protobuf/minprotobuf
 import libp2p/utility
 import libp2p/errors
-import libp2p/crypto/[crypto, chacha20poly1305]
+import libp2p/crypto/[crypto, chacha20poly1305, curve25519]
+
 
 logScope:
   topics = "wakunoise"
 
+#################################################################
+
+# Constants and data structures
+
 const
-  # EmptyKey is a special value which indicates a ChaChaPolyKey has not yet been initialized.
+  # EmptyKey represents a non-initialized ChaChaPolyKey
   EmptyKey = default(ChaChaPolyKey)
+  # The maximum ChaChaPoly allowed nonce in Noise Handshakes
+  NonceMax = uint64.high - 1
 
 type
+  # Default underlying elliptic curve arithmetic (useful for switching to multiple ECs)
+  # Current default is Curve25519
+  EllipticCurveKey = Curve25519Key
+
+  # An EllipticCurveKey (public, private) key pair
+  KeyPair* = object
+    privateKey: EllipticCurveKey
+    publicKey: EllipticCurveKey
+
+  # A Noise public key is a public key exchanged during Noise handshakes (no private part)
+  # This follows https://rfc.vac.dev/spec/35/#public-keys-serialization
+  # pk contains the X coordinate of the public key, if unencrypted (this implies flag = 0)
+  # or the encryption of the X coordinated concatenated with the authorization tag, if encrypted (this implies flag = 1)
```

Comment:
```suggestion
  # or the ChaChaPoly encryption of the X coordinate concatenated with the authorization tag, if encrypted (this implies flag = 1)
```

Reply:
I prefer to not specifically refer to underlying primitives, when possible, since we may add support to AES-GCM in the future (and also Blake and Curve448).